corehq/apps/analytics/tasks.py

```diff
@@ -234,19 +228,22 @@ def _send_form_to_hubspot(form_id, webuser, hubspot_cookie, meta, extra_fields=N
                 'firstname': webuser.first_name,
                 'lastname': webuser.last_name,
             })
+        if extra_fields:
+            data.update(extra_fields)
 
-        response = _send_hubspot_form_request(url, data)
+        response = _send_hubspot_form_request(hubspot_id, form_id, data)
         _log_response('HS', data, response)
         response.raise_for_status()
 
-        # these must be submitted after the form to ensure
-        # the contact has been created in hubspot
-        if extra_fields:
-            update_hubspot_properties_v2(webuser, extra_fields)
-
 
 @count_by_response_code(DATADOG_HUBSPOT_SENT_FORM_METRIC)
-def _send_hubspot_form_request(url, data):
+def _send_hubspot_form_request(hubspot_id, form_id, data):
```

Comment:
out of curiosity, why did you move this down here?

Reply:
Seemed like a more appropriate place for it.  This way the function above doesn't need the URL, it just says "send this form with this data", without needing to worry about the details of assembling the URL and initializing the request.

 Plus, the URL used for this endpoint determines what parameters need to be passed to the request.  For instance, I attempted upgrading to v3 (more on that in the PR description), which would mean that `data` should be passed as JSON rather than HTTP form data.